### PR TITLE
SUS-3255 | document FLAG_* constants in WikiFactory

### PR DESCRIPTION
**Pull request by macbre**

### CODE REVIEW

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 1]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
 	];
 
 	# city_flags
-	const FLAG_CREATE_DB_DUMP        = 1;
-	const FLAG_CREATE_IMAGE_ARCHIVE  = 2;
-	const FLAG_DELETE_DB_IMAGES      = 4;
-	const FLAG_FREE_WIKI_URL         = 8;
-	const FLAG_HIDE_DB_IMAGES        = 16;
-	const FLAG_REDIRECT              = 32;
+	const FLAG_CREATE_DB_DUMP        = 1; // prepare a database dump and send it to Amazon's s3
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:65]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 2]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
+	const FLAG_CREATE_IMAGE_ARCHIVE  = 2; // prepare DFS bucket dump and send it to Amazon's s3
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:66]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 3]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
+	const FLAG_DELETE_DB_IMAGES      = 4; // remove DFS bucket
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:67]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 4]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
+	const FLAG_FREE_WIKI_URL         = 8; // removes wiki database and WikiFactory settings
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:68]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 5]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
+	const FLAG_HIDE_DB_IMAGES        = 16;  // images and DB dumps will be hidden on s3
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:69]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 6]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
+	const FLAG_REDIRECT              = 32;  // this wiki is a redirect - do not remove
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:70]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 7]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
 	const FLAG_ADOPTABLE             = 64;  //used by AutomaticWikiAdoption
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:71]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 8]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
 	const FLAG_ADOPT_MAIL_FIRST      = 128; //used by AutomaticWikiAdoption
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:72]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 9]
```diff
@@ -62,12 +62,12 @@ class WikiFactory {
...
 	const FLAG_ADOPT_MAIL_SECOND     = 256; //used by AutomaticWikiAdoption
```
Comment (WikiaTech) [inline extensions/wikia/WikiFactory/WikiFactory.php:73]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 120 issues
* ![MAJOR][MAJOR] 87 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 19 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiFactory.php#](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#): File "WikiFactory.php" has 3465 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WikiFactory.php#L27](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L27): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L49](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L49): Class "WikiFactory" has 77 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiFactory.php#L255](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L255): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [WikiFactory.php#L259](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L259): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L264](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L264): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L358](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L358): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L404](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L404): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L487](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L487): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiFactory.php#L568](https://github.com/Wikia/app/blob/95ec1cc60118b25bdf594e4029cb1b1a3c639bb2/extensions/wikia/WikiFactory/WikiFactory.php#L568): The Cyclomatic Complexity of this function "setVarById" is 30 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**